server/src/main/java/org/elasticsearch/http/DefaultRestChannel.java

```diff
@@ -171,4 +165,13 @@ private boolean isCloseConnection() {
     private boolean isHttp10() {
         return request.getHttpRequest().protocolVersion() == HttpRequest.HttpVersion.HTTP_1_0;
     }
+
+    // Determine if the request method is HEAD, swallowing any exception if the request method is invalid
+    private boolean isHead() {
+        try {
+            return RestRequest.Method.HEAD == request.method();
+        } catch (Exception ignored) {
+            return false;
```

Comment:
Maybe pass the response into this method and assert that it's an error response via its status? (I think that would make it easier to understand why this is ok to swallow). Technically I guess this should have status `405` (method not allowed?)

Reply:
I pushed a64c1896d4695d4ec28cec57044414435d28596d